cmd/agent/subcommands/run/command_notwin.go

```diff
@@ -0,0 +1,59 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build !windows
+
+// Package run implements 'agent run' (and deprecated 'agent start').
+package run
+
+import (
+	_ "expvar"         // Blank import used because this isn't directly used in this file
+	_ "net/http/pprof" // Blank import used because this isn't directly used in this file
+
+	// core components
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/comp/core/flare"
+	"github.com/DataDog/datadog-agent/comp/core/log"
+	"github.com/DataDog/datadog-agent/comp/core/sysprobeconfig"
+	"github.com/DataDog/datadog-agent/comp/core/telemetry"
+	"github.com/DataDog/datadog-agent/comp/dogstatsd/replay"
+	dogstatsdServer "github.com/DataDog/datadog-agent/comp/dogstatsd/server"
+	dogstatsdDebug "github.com/DataDog/datadog-agent/comp/dogstatsd/serverDebug"
+	"github.com/DataDog/datadog-agent/comp/forwarder/defaultforwarder"
+	logsAgent "github.com/DataDog/datadog-agent/comp/logs/agent"
+	"github.com/DataDog/datadog-agent/comp/metadata/runner"
+	otelcollector "github.com/DataDog/datadog-agent/comp/otelcol/collector"
+	"github.com/DataDog/datadog-agent/comp/remote-config/rcclient"
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/serializer"
+	"github.com/DataDog/datadog-agent/pkg/util"
+	"go.uber.org/fx"
+	// runtime init routines
+)
+
+func run(log log.Component,
+	config config.Component,
+	flare flare.Component,
+	telemetry telemetry.Component,
+	sysprobeconfig sysprobeconfig.Component,
+	server dogstatsdServer.Component,
+	capture replay.Component,
+	serverDebug dogstatsdDebug.Component,
+	forwarder defaultforwarder.Component,
+	rcclient rcclient.Component,
+	metadataRunner runner.Component,
+	demux *aggregator.AgentDemultiplexer,
+	sharedSerializer serializer.MetricSerializer,
+	cliParams *cliParams,
+	logsAgent util.Optional[logsAgent.Component],
+	otelcollector otelcollector.Component,
+) error {
+	return commonRun(log, config, flare, telemetry, sysprobeconfig, server, capture, serverDebug, forwarder, rcclient, metadataRunner, demux, sharedSerializer, cliParams, logsAgent, otelcollector)
```

Comment:
On linux, there is no difference between run & commonRun
On windows, there's a difference in the signature of run.  It accepts additional components not used/avail in linux

Reply:
Can you implement the solution we discussed during the office hour, ie, having one `newAgentCrashComponent` for non windows platform that just return the component without starting it, and one `newAgentCrashComponent` for windows that creates and starts the component? 
It will make the code much more easier as you can have a single run function for all platform.

Reply:
I think that makes it more confusing for linux, having components loaded that aren't used.  it will also load the tarce agent config cmponent, which wouldn't be used in linux, so it would be needlessly instantiating an actual component.